subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolutionstrategy/DefaultCachePolicy.java

```diff
@@ -38,6 +38,7 @@
     final List<Action<? super DependencyResolutionControl>> dependencyCacheRules;
     final List<Action<? super ModuleResolutionControl>> moduleCacheRules;
     final List<Action<? super ArtifactResolutionControl>> artifactCacheRules;
+    private final List<Runnable> mutateActions = new ArrayList<Runnable>();
```

Comment:
Used a `List` here instead of the usual `LinkedHashSet` for performance reasons. This is most probably only going to contain one item at a time, and `Set`'s uniqueness would not add much of a benefit to warrant its much higher cost.